source/slang/slang-parser.cpp

```diff
@@ -4364,6 +4382,46 @@ namespace Slang
         return attrDecl;
     }
 
+    static void addSpecialGLSLModifiersBasedOnType(
+        Parser* parser,
+        Decl* decl,
+        Modifiers* modifiers)
+    {
+        // we intentionally have findModifier twice since GLSLOffsetLayoutAttribute is rare
+        // better to early leave
+        if (auto bindingMod = modifiers->findModifier<GLSLBindingAttribute>())
+        {
+            if (auto varDeclBase = as<VarDeclBase>(decl))
+            {
+                if (auto declRefExpr = as<DeclRefExpr>(varDeclBase->type.exp))
+                {
+                    // here is a problem; we link types into a literal in IR stage post parse
+                    // but, order (top down) mattter when parsing atomic_uint offset
+                    // more over, we can have patterns like: offset = 20, no offset [+4], offset = 16.
+                    // Therefore we must parse all in order. The issue then is will struggle to 
+                    // subsitute atomic_uint for storage buffers...
+                    if (auto name = declRefExpr->name)
+                    {
+                        if (name->text.equals("atomic_uint"))
+                        {
+                            if (!modifiers->findModifier<GLSLOffsetLayoutAttribute>())
+                            {
+                                const int64_t nextOffset = parser->getNextBindingOffset(bindingMod->binding);
+                                GLSLOffsetLayoutAttribute* modifier = parser->astBuilder->create<GLSLOffsetLayoutAttribute>();
+                                modifier->keywordName = NULL; //no keyword name given
+                                modifier->loc = bindingMod->loc; //has no location in file, set to parent binding
+                                modifier->offset = nextOffset;
+                                
+                                Modifiers newModifier;
+                                newModifier.first = modifier;
+                                _addModifiers(decl, newModifier);
```

Comment:
Do we have a test coverage for this code and the examples mentioned in the comment?
If not, can you add some?